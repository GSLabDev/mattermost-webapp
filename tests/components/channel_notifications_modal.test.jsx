import React from 'react';
import {shallow} from 'enzyme';
import {Modal} from 'react-bootstrap';

import {mountWithIntl} from 'tests/helpers/intl-test-helper';

import ChannelNotificationsModal from 'components/channel_notifications_modal/channel_notifications_modal.jsx';

describe('components/channel_notifications_modal/ChannelNotificationsModal', () => {
    const channelMember = {
        allow_marketing: true,
        auth_data: '',
        auth_service: '',
        create_at: 1511180034368,
        delete_at: 0,
        email: 'email@email.com',
        first_name: '',
        id: 'xmwtt3ag6jdujcndz17s3szcah',
        last_name: '',
        last_password_update: 1511180034368,
        locale: 'en',
        nickname: '',
        notify_props: {
            channel: 'true',
            desktop: 'all',
            desktop_sound: 'true',
            email: 'true',
            first_name: 'false',
            mention_keys: 'username,@username',
            push: 'mention'
        },
        position: '',
        roles: 'system_admin system_user',
        update_at: 1511206855215,
        username: 'username'
    };

    test('should match snapshot', () => {
        function emptyFunction() {} //eslint-disable-line no-empty-function
        global.window.mm_config = {};

        const wrapper = shallow(
            <ChannelNotificationsModal
                show={true}
                onHide={emptyFunction}
                channel={{name: 'testchannel', displayName: 'testchannel', header: '', purpose: ''}}
                channelMember={channelMember}
                currentUser={{id: 'current_user_id'}}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should call onHide callback when modal is hidden', (done) => {
        function onHide() {
            done();
        }

        const wrapper = mountWithIntl(
            <ChannelNotificationsModal
                show={true}
                onHide={onHide}
                channel={{name: 'testchannel', displayName: 'testchannel', header: '', purpose: ''}}
                channelMember={channelMember}
                currentUser={{id: 'current_user_id'}}
            />
        );

        wrapper.find(Modal).first().props().onExited();
    });
});