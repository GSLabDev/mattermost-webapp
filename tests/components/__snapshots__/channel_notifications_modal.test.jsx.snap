// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/channel_notifications_modal/ChannelNotificationsModal should match snapshot 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="settings-modal settings-modal--tabless"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onExited={[Function]}
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      <FormattedMessage
        defaultMessage="Notification Preferences for "
        id="channel_notifications.preferences"
        values={Object {}}
      />
      <span
        className="name"
      />
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div
      className="settings-table"
    >
      <div
        className="settings-content"
      >
        <div
          className="user-settings"
        >
          <br />
          <div
            className="divider-dark first"
          />
          <SettingItemMin
            describe={
              <FormattedMessage
                id="channel_notifications.never"
                values={Object {}}
              />
            }
            title={
              <FormattedMessage
                defaultMessage="Send desktop notifications"
                id="channel_notifications.sendDesktop"
                values={Object {}}
              />
            }
            updateSection={[Function]}
          />
          <div>
            <div
              className="divider-light"
            />
            <SettingItemMin
              describe={
                <FormattedMessage
                  id="channel_notifications.never"
                  values={Object {}}
                />
              }
              title={
                <FormattedMessage
                  defaultMessage="Send mobile push notifications"
                  id="channel_notifications.push"
                  values={Object {}}
                />
              }
              updateSection={[Function]}
            />
          </div>
          <div
            className="divider-light"
          />
          <SettingItemMin
            describe={
              <FormattedMessage
                defaultMessage="For all unread messages"
                id="channel_notifications.allUnread"
                values={Object {}}
              />
            }
            title={
              <FormattedMessage
                defaultMessage="Mark Channel Unread"
                id="channel_notifications.markUnread"
                values={Object {}}
              />
            }
            updateSection={[Function]}
          />
          <div
            className="divider-dark"
          />
        </div>
      </div>
    </div>
  </ModalBody>
</Modal>
`;
